torch/distributed/optim/zero_redundancy_optimizer.py

```diff
@@ -460,39 +710,97 @@ def _device_to_per_rank_params(self) -> Dict[torch.device, List[List[torch.Tenso
             ...
         ...
         """
-        if len(self._device_to_per_rank_params_cache) == 0:
+        assert self.parameters_as_bucket_view, \
+            "`_device_to_params_per_rank` should only be used if " \
+            "`parameters_as_bucket_view=True`"
+        if len(self._device_to_params_per_rank_cache) == 0:
             for rank, param_groups in enumerate(self._partition_parameters()):
                 for param_group in param_groups:
                     for param in param_group["params"]:
                         device = param.device
-                        if device not in self._device_to_per_rank_params_cache:
-                            self._device_to_per_rank_params_cache[device] = [[] for _ in range(self.world_size)]
-                        self._device_to_per_rank_params_cache[device][rank].append(param)
-        return self._device_to_per_rank_params_cache
+                        if device not in self._device_to_params_per_rank_cache:
+                            self._device_to_params_per_rank_cache[device] = [[] for _ in range(self.world_size)]
+                        self._device_to_params_per_rank_cache[device][rank].append(param)
+        return self._device_to_params_per_rank_cache
 
-    def step(
+    @property
+    def _device_to_params(
```

Comment:
curious, why this is `device_to_params` instead of `device_to_buckets`?

Reply:
Good point -- I will change it. It was from before I created the class `_DDPBucket` and instead had a `List[torch.Tensor]` as the inner-most element in the `dict` value.